controllers/managedocs_controller.go

```diff
@@ -1482,3 +1518,18 @@ func (r *ManagedOCSReconciler) unrestrictedGet(obj client.Object) error {
 	key := client.ObjectKeyFromObject(obj)
 	return r.UnrestrictedClient.Get(r.ctx, key, obj)
 }
+
+func (r *ManagedOCSReconciler) ensureMCGDeployment(storageCluster *ocsv1.StorageCluster, enableMCGAsString string) error {
+	// Check and enable MCG in Storage Cluster spec
+	mcgEnable, err := strconv.ParseBool(enableMCGAsString)
+	if err != nil {
+		return fmt.Errorf("Invalid Enable MCG value: %v", enableMCGAsString)
+	}
```

Comment:
This is a util function that has the concern to ensure MCG's desired state. 
The concern of parsing the MCG flag does not belong here, and in any case, is a single line operation. 
Please move it to the call site and change the sig of the function to accept an `enableMCG bool` instead